Ska/Matplotlib/core.py

```diff
@@ -163,15 +164,23 @@ def cxctime2plotdate(times):
     Convert input CXC time (sec) to the time base required for the matplotlib
     plot_date function (days since start of year 1).
 
-    :param times: iterable list of times
+    :param times: times (any DateTime compatible format or object)
     :rtype: plot_date times
     """
+    # Convert times to float array of CXC seconds
+    if isinstance(times, (DateTime, CxoTime)):
+        times = times.secs
+    else:
+        times = np.asarray(times)
+
+        # If not floating point then use CxoTime to convert to seconds
```

Comment:
Just curious--what does this catch that the above doesn't?

Reply:
I'm not 100% sure on what you mean by `this` and `the above`. But a `list` or `np.array` of strings will go through the `CxoTime(times).secs` line below.  A list or array of float numbers is assumed to be CXC seconds. As a slightly odd case, a list of `CxoTime` objects will have `dtype.kind == 'O'` and fall through and get turned correctly into seconds.